community/kernel/src/main/java/org/neo4j/kernel/impl/util/ArrayQueueOutOfOrderSequence.java

```diff
@@ -96,6 +99,24 @@ public long highestEverSeen()
         return createResult( number, meta );
     }
 
+    @Override
+    public synchronized void await( long awaitedNumber, long timeoutMillis ) throws TimeoutException, InterruptedException
+    {
+        long endTime = System.currentTimeMillis() + timeoutMillis;
```

Comment:
Although prudent, in practice someone would have to enter a timeout in excess of 290 million years for that to be a problem.